packages/core/src/languages/solidity/compiler.ts

```diff
@@ -104,3 +108,37 @@ export const getMinimumCompilerInput = (
 
   return minimumCompilerInput
 }
+
+/**
+ * Returns the minimum compiler output for a given source name.
+ *
+ * @param fullCompilerOutput The full compiler output object.
+ * @param fullOutputContracts The full compiler output contract object.
+ * @param sourceName The source name.
+ * @returns Minimum compiler input necessary to compile the source name.
+ */
+export const getMinimumCompilerOutput = (
+  fullCompilerOutput: CompilerOutput,
+  fullOutputContracts: CompilerOutputContracts,
+  sourceName: string,
+  contractName: string
+): CompilerOutput => {
+  const contractOutput = fullOutputContracts[sourceName][contractName]
+  const metadata: CompilerOutputMetadata =
+    typeof contractOutput.metadata === 'string'
+      ? JSON.parse(contractOutput.metadata)
+      : contractOutput.metadata
+
+  const minimumSources: CompilerOutput['sources'] = {}
+  const minimumContracts: CompilerOutput['contracts'] = {}
+  for (const newSourceName of Object.keys(metadata.sources)) {
+    minimumSources[newSourceName] = fullCompilerOutput.sources[newSourceName]
+    minimumContracts[newSourceName] =
+      fullCompilerOutput.contracts[newSourceName]
+  }
+
+  fullCompilerOutput.contracts = minimumContracts
+  fullCompilerOutput.sources = minimumSources
```

Comment:
I don't think we should overwrite these members of `fullCompilerOutput`, since we may use them elsewhere. Instead, I think we should make a new object like in `getMInimumCompilerInput`